src/realm/table.cpp

```diff
@@ -333,6 +333,72 @@ void LinkChain::add(ColKey ck)
     m_link_cols.push_back(ck);
 }
 
+LinkTranslator::LinkTranslator(Obj origin, ColKey origin_col_key)
```

Comment:
I would have this defined into a separate header/source file, this is basically a basic class to deal with links from now on..